### ISSUE TITLE
Create NPC dialog box - V2

### ISSUE THREAD

**Issue by tiny-crab** [zakattak/Timbre#5]
### Description:
With V1, the player can read the NPCs' messages in a single dialog box at the bottom of the screen. A few minor changes can be made to make it slightly better.

Making the message "write" itself dynamically with a word or character at a time will help players read-along with it.

Putting the dialog box above the NPC speaking the message will be a good challenge as well.

### Definition of Done:
- Words/characters in the message appear one by one at a static speed.
- The dialog box now appears over the head of the NPC, not at the bottom of the screen.
- Convert dialog boxes to use the ControllerThread and ControllerInteractable classes, and freeze main character input while an NPC is talking.

**Comment by tiny-crab** [comment]
I created a couple of sleeper classes that don't do anything yet:
* https://github.com/zakattak/Timbre/blob/master/Assets/Scripts/ControllerThread.cs
* https://github.com/zakattak/Timbre/blob/master/Assets/Scripts/ControllerInteractable.cs

My idea was that it would be used to "focus" the controls to a specific entity in the game. For example:
* The player would use WASD to move the main character up to an NPC, and press F to interact. The MC is the "focused" ControllerInteractable object.
* The NPC would begin spouting dialogue boxes at the player. To continue through each dialog box, the player would press F. The dialog box is the "focused" ControllerInteractable object, and WASD would no longer work to control the player.

This is a fairly simple concept, but if we route all player controls through the ControllerThread object, it might help to avoid "pausing" any given object in the screen while the player interacts with it. Instead of having some `inDialog` bool in the player controller for the Dialog state, and adding in `inInventory`, or `inFooState` branches later, then we would define the `Inventory`, `Dialog`, and `FooState` controls in their own object, and the ControllerThread would pick the state based on the currently focused ControllerInteractable.

I made a comment in the ControllerThread object that is good to copy here:
```
// Let's imagine that the player is just another thread running in the game.
// There will only be one thing they can "execute" at once, so let's force the interactions
// between the keyboard and game to be limited to a single object.
```

With this issue, it would **also** be good to input multi-screen dialog boxes that become "focused" when the player interacts with an NPC. Then the player can press F (or any other key) to scroll through the dialog.
